radvd.c

```diff
@@ -573,7 +573,24 @@ static struct Interface *main_loop(int sock, struct Interface *ifaces, char cons
 
 		if (sighup_received) {
 			dlog(LOG_INFO, 3, "sig hup received");
-			ifaces = reload_config(sock, ifaces, conf_path);
+
+			/*
+			 * check that 'other' cannot write the file
+			 * for non-root, also that self/own group can't either
+			 */
+			int do_reload = 1;
+			if (check_conffile_perm(username, conf_path) != 0) {
+				if (get_debuglevel() == 0) {
+					flog(LOG_ERR, "permissions on conf_file invalid, ignoring reload request");
+					do_reload = 0;
+				} else {
+					flog(LOG_WARNING, "Insecure file permissions, but continuing reload anyway");
+				}
+			}
+
+			if (do_reload) {
+				ifaces = reload_config(sock, ifaces, conf_path);
+			}
```

Comment:
this just almost-duplicates the  existing code. 

Please refactor to introduce a new function that gets used in both places, and either exits or skips reloading in the caller.